File: TellDontAsk/AlarmsObjectOriented/Building.cs
using AlarmsObjectOriented.Interfaces;

namespace AlarmsObjectOriented
{
  public class Building
  {
    private readonly Alarm _alarm;

    public Building(Alarm alarm)
    {
      _alarm = alarm;
    }

    public void SomeoneEntered()
    {
      _alarm.Trigger();
    }

    public void AllClear()
    {
      _alarm.Disable();
    }

    public void Dump()
    {
      _alarm.Dump();
    }
  }

}

File: Java/AlarmsObjectOriented/src/main/java/Building.java
import interfaces.Alarm;

public class Building {
  private final Alarm _alarm;

  public Building(Alarm alarm) {
    _alarm = alarm;
  }

  public void someoneEntered() {
    _alarm.trigger();
  }

  public void allClear() {
    _alarm.disable();
  }

  public void dump() {
    _alarm.dump();
  }
}

